Source/WebKit/UIProcess/API/glib/WebKitWebExtensionMatchPattern.cpp

```diff
@@ -0,0 +1,441 @@
+    {
+    }
+#endif
+};
+
+G_DEFINE_BOXED_TYPE(WebKitWebExtensionMatchPattern, webkit_web_extension_match_pattern, webkit_web_extension_match_pattern_ref, webkit_web_extension_match_pattern_unref)
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+/**
+ * webkit_web_extension_match_pattern_register_custom_URL_scheme:
+ * @urlScheme: The custom URL scheme to register
+ *
+ * Registers a custom URL scheme that can be used in match patterns.
+ * 
+ * This method should be used to register any custom URL schemes used by the app for the extension base URLs,
+ * other than `webkit-extension`, or if extensions should have access to other supported URL schemes when using `<all_urls>`.
+ */
+void webkit_web_extension_match_pattern_register_custom_URL_scheme(const gchar* urlScheme)
+{
+    g_return_if_fail(WTF::URLParser::maybeCanonicalizeScheme(String::fromUTF8(urlScheme)));
+
+    WebKit::WebExtensionMatchPattern::registerCustomURLScheme(String::fromUTF8(urlScheme));
+}
+
+WebKitWebExtensionMatchPattern* webkitWebExtensionMatchPatternCreate(const RefPtr<WebExtensionMatchPattern>& apiMatchPattern)
+{
+    if (!apiMatchPattern)
+        return nullptr;
+
+    ASSERT(API::Object::unwrap(static_cast<void*>(apiMatchPattern.get()))->type() == API::Object::Type::WebExtensionMatchPattern);
+    WebKitWebExtensionMatchPattern* matchPattern = static_cast<WebKitWebExtensionMatchPattern*>(fastMalloc(sizeof(WebKitWebExtensionMatchPattern)));
+    new (matchPattern) WebKitWebExtensionMatchPattern(static_pointer_cast<WebExtensionMatchPattern>(apiMatchPattern));
+    return matchPattern;
+}
+
+WebKitWebExtensionMatchPattern* webkitWebExtensionMatchPatternCreate(Ref<WebExtensionMatchPattern>& matchPattern)
+{
+    RefPtr<WebExtensionMatchPattern> apiMatchPattern = adoptRef(matchPattern.get());
+
+    WebKitWebExtensionMatchPattern* pattern = static_cast<WebKitWebExtensionMatchPattern*>(fastMalloc(sizeof(WebKitWebExtensionMatchPattern)));
+    new (pattern) WebKitWebExtensionMatchPattern(static_pointer_cast<WebExtensionMatchPattern>(apiMatchPattern));
+    return pattern;
+}
+
+/**
+ * webkit_web_extension_match_pattern_ref:
+ * @matchPattern: a #WebKitWebExtensionMatchPattern
+ *
+ * Atomically acquires a reference on the given @matchPattern.
+ *
+ * This function is MT-safe and may be called from any thread.
+ *
+ * Returns: The same @matchPattern with an additional reference.
+ */
+WebKitWebExtensionMatchPattern* webkit_web_extension_match_pattern_ref(WebKitWebExtensionMatchPattern* matchPattern)
+{
+    g_return_val_if_fail(matchPattern, nullptr);
+    g_atomic_int_inc(&matchPattern->referenceCount);
+    return matchPattern;
+}
+
+/**
+ * webkit_web_extension_match_pattern_unref:
+ * @matchPattern: a #WebKitWebExtensionMatchPattern
+ *
+ * Atomically releases a reference on the given @matchPattern.
+ *
+ * If the reference was the last, the resources associated to the
+ * @matchPattern are freed. This function is MT-safe and may be called from
+ * any thread.
+ */
```

Comment:
```suggestion
 * any thread.
 *
 * Since: 2.48
 */
```

I will stop adding suggestions for the Since tags. Just add them everywhere please!